### PR TITLE
[CONSUL-186] Replace Unsupported Commands

### PR DESCRIPTION
**Pull request by joselo85**
To test this file you need Docker containers running with the following tags:

- consul-dev
- bats-verify

You can run the build images script by navigating in your terminal to its location and issuing the following command:
`bash build-images.windows.sh`

### CODE REVIEW

**Comment by joselo85** [comment]
To test this file you need Docker containers running with the following tags:

- consul-dev
- bats-verify

You can run the build images script by navigating in your terminal to its location and issuing the following command:
`bash build-images.windows.sh`

File: test/integration/connect/envoy/run-tests.windows.sh [hunk 1]
```diff
@@ -68,7 +68,7 @@ function init_workdir {
   # Copy all the test files
   find ${CASE_DIR} -maxdepth 1 -name '*.bats' -type f -exec cp -f {} workdir/${CLUSTER}/bats \;
   # Copy CLUSTER specific bats
-  cp helpers.bash workdir/${CLUSTER}/bats
+  cp helpers.windows.bash workdir/${CLUSTER}/bats
```
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh:70]:
```suggestion
  cp helpers.windows.bash workdir/${CLUSTER}/bats/helpers.bash
```

File: test/integration/connect/envoy/run-tests.windows.sh [hunk 2]
```diff
@@ -413,11 +413,12 @@ function global_setup {
...
 
 function wipe_volumes {
-  docker run --rm -i \
+  docker.exe run --rm -i \
     $WORKDIR_SNIPPET \
     --net=none \
-    "${HASHICORP_DOCKER_PROXY}/alpine" \
-    sh -c 'rm -rf /workdir/*'
+    # TODO: Temporary change, until HashiCorp updates HASHICORP_DOCKER_PROXY.
+    "${HASHICORP_DOCKER_PROXY}/windows/nanoserver" \
```
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh]:
```suggestion
    "${HASHICORP_DOCKER_PROXY}/windows/nanoserver" \
```

File: test/integration/connect/envoy/run-tests.windows.sh [hunk 3]
```diff
@@ -513,47 +514,50 @@ function test_teardown {
...
     # This is a dummy container that we use to create volume and keep it
     # accessible while other containers are down.
-    docker volume create envoy_workdir &>/dev/null
-    docker run -d --name envoy_workdir_1 \
+    docker.exe volume create envoy_workdir &>/dev/null
+    docker.exe run -d --name envoy_workdir_1 \
         $WORKDIR_SNIPPET \
         --net=none \
-        k8s.gcr.io/pause &>/dev/null
+        mcr.microsoft.com/windows/nanoserver:1809 &>/dev/null
```
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh:533]:
```suggestion
        "${HASHICORP_DOCKER_PROXY}/windows/nanoserver" &>/dev/null
```

File: test/integration/connect/envoy/run-tests.windows.sh [hunk 4]
```diff
@@ -513,47 +514,50 @@ function test_teardown {
...
     # TODO(rb): switch back to "${HASHICORP_DOCKER_PROXY}/google/pause" once that is cached
 
     # pre-build the verify container
     echo "Rebuilding 'bats-verify' image..."
-    docker build -t bats-verify -f Dockerfile-bats .
+    # TODO -Line below commented for testing
+    # docker build -t bats-verify -f Dockerfile-bats-windows .
```
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh:539]:
```suggestion
    docker.exe build -t bats-verify -f Dockerfile-bats-windows .
```

File: test/integration/connect/envoy/run-tests.windows.sh [hunk 5]
```diff
@@ -513,47 +514,50 @@ function test_teardown {
...
 
     # if this fails on CircleCI your first thing to try would be to upgrade
     # the machine image to the latest version using this listing:
     #
     # https://circleci.com/docs/2.0/configuration-reference/#available-linux-machine-images
     echo "Checking bats image..."
-    docker run --rm -t bats-verify -v
+    # TODO - Line below commented for testing
+    docker.exe run --rm -t bats-verify -v
```
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh:547]:
```suggestion
    docker.exe run --rm -t bats-verify -v
```

File: test/integration/connect/envoy/run-tests.windows.sh [hunk 6]
```diff
@@ -591,10 +595,10 @@ function common_run_container_service {
   local httpPort="$3"
   local grpcPort="$4"
 
-  docker run -d --name $(container_name_prev) \
+  docker.exe run -d --name $(container_name_prev) \
     -e "FORTIO_NAME=${service}" \
-    $(network_snippet $CLUSTER) \
-    "${HASHICORP_DOCKER_PROXY}/fortio/fortio" \
+    $(network_snippet $CLUSTER) \    
+   "${HASHICORP_DOCKER_PROXY}/windows/fortio" \
```
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh]:
```suggestion
    $(network_snippet $CLUSTER) \
   "${HASHICORP_DOCKER_PROXY}/windows/fortio" \
```

File: test/integration/connect/envoy/run-tests.windows.sh [hunk 7]
```diff
@@ -784,32 +788,33 @@ function run_container_fake-statsd {
...
     "${HASHICORP_DOCKER_PROXY}/openzipkin/zipkin"
 }
-
+# TODO
 function run_container_jaeger {
-  docker run -d --name $(container_name) \
+  docker.exe run -d --name $(container_name) \
     $WORKDIR_SNIPPET \
     $(network_snippet primary) \
     "${HASHICORP_DOCKER_PROXY}/jaegertracing/all-in-one:1.11" \
```
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh:808]:
```suggestion
    "${HASHICORP_DOCKER_PROXY}/windows/jaeger" \
```

File: test/integration/connect/envoy/run-tests.windows.sh [hunk 8]
```diff
@@ -784,32 +788,33 @@ function run_container_fake-statsd {
...
 #https://hub.docker.com/r/openzipkin/zipkin-base
 function run_container_zipkin {
-  docker run -d --name $(container_name) \
+  docker.exe run -d --name $(container_name) \
     $WORKDIR_SNIPPET \
     $(network_snippet primary) \
     "${HASHICORP_DOCKER_PROXY}/openzipkin/zipkin"
 }
-
+# TODO
```
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh]:
```suggestion

```

File: test/integration/connect/envoy/run-tests.windows.sh [hunk 9]
```diff
@@ -784,32 +788,33 @@ function run_container_fake-statsd {
...
 }
 
+# TODO
 #https://hub.docker.com/r/openzipkin/zipkin-base
 function run_container_zipkin {
-  docker run -d --name $(container_name) \
+  docker.exe run -d --name $(container_name) \
     $WORKDIR_SNIPPET \
     $(network_snippet primary) \
     "${HASHICORP_DOCKER_PROXY}/openzipkin/zipkin"
```
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh:802]:
```suggestion
    "${HASHICORP_DOCKER_PROXY}/windows/zipkin"
```

File: test/integration/connect/envoy/run-tests.windows.sh [hunk 10]
```diff
@@ -784,32 +788,33 @@ function run_container_fake-statsd {
...
+  docker.exe run -d --name $(container_name) \
     $WORKDIR_SNIPPET \
     $(network_snippet primary) \
-    "${HASHICORP_DOCKER_PROXY}/alpine/socat:1.7.3.4-r1" \
+    "${HASHICORP_DOCKER_PROXY}/windows/socat" \
     -u UDP-RECVFROM:8125,fork,reuseaddr \
     SYSTEM:'xargs -0 echo >> /workdir/primary/statsd/statsd.log'
 }
 
+# TODO
```
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh]:
```suggestion

```

File: test/integration/connect/envoy/run-tests.windows.sh [hunk 11]
```diff
@@ -849,10 +854,10 @@ function common_run_container_tcpdump {
     local DC="$1"
 
     # we cant run this in circle but its only here to temporarily enable.
+    # TODO
```
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh]:
```suggestion

```

File: test/integration/connect/envoy/run-tests.windows.sh [hunk 12]
```diff
@@ -513,47 +514,50 @@ function test_teardown {
...
 
     # pre-build the consul+envoy container
     echo "Rebuilding 'consul-dev-envoy:${ENVOY_VERSION}' image..."
-    docker build -t consul-dev-envoy:${ENVOY_VERSION} \
-        --build-arg ENVOY_VERSION=${ENVOY_VERSION} \
-        -f Dockerfile-consul-envoy .
+    # TODO - Line below commented for testing
+    # docker build -t consul-dev-envoy:${ENVOY_VERSION} \
+    #     --build-arg ENVOY_VERSION=${ENVOY_VERSION} \
+    #     -f Dockerfile-consul-envoy .
```
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh:554]:
```suggestion
    docker build -t consul-dev-envoy:${ENVOY_VERSION} \
        --build-arg ENVOY_VERSION=${ENVOY_VERSION} \
        -f Dockerfile-consul-envoy-windows .
```

**Review by ezfepo — CHANGES_REQUESTED** [review]

File: test/integration/connect/envoy/run-tests.windows.sh [hunk 13]
```diff
@@ -37,7 +37,7 @@ function command_error {
 }
 
 trap 'command_error $? "${BASH_COMMAND}" "${LINENO}" "${FUNCNAME[0]:-main}" "${BASH_SOURCE[0]}:${BASH_LINENO[0]}"' ERR
-
+# TODO
```
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh]:
```suggestion

```

File: test/integration/connect/envoy/run-tests.windows.sh [hunk 14]
```diff
@@ -296,7 +296,7 @@ function pre_service_setup {
 function start_services {
   # Push the state to the shared docker volume (note this is because CircleCI
   # can't use shared volumes)
-  docker cp workdir/. envoy_workdir_1:/workdir
+ docker.exe cp workdir/. envoy_workdir_1:/workdir
```
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh]:
```suggestion
  docker.exe cp workdir/. envoy_workdir_1:/workdir
```

File: test/integration/connect/envoy/run-tests.windows.sh (hunk 2 above)
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh]:
```suggestion
    "${HASHICORP_DOCKER_PROXY}/windows/nanoserver" \
```

File: test/integration/connect/envoy/run-tests.windows.sh (hunk 6 above)
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh]:
```suggestion
    $(network_snippet $CLUSTER) \
   "${HASHICORP_DOCKER_PROXY}/windows/fortio" \
```

File: test/integration/connect/envoy/run-tests.windows.sh (hunk 10 above)
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh]:
```suggestion

```

File: test/integration/connect/envoy/run-tests.windows.sh (hunk 8 above)
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh]:
```suggestion

```

File: test/integration/connect/envoy/run-tests.windows.sh (hunk 11 above)
Comment (ezfepo) [inline test/integration/connect/envoy/run-tests.windows.sh]:
```suggestion

```

**Review by ezfepo — CHANGES_REQUESTED** [review]

**Review by ezfepo — APPROVED** [review]

**Pull request merged**